packages/contracts-bedrock/contracts/dispute/lib/LibPosition.sol

```diff
@@ -0,0 +1,121 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+import "../../libraries/DisputeTypes.sol";
+
+/**
+ * @title LibPosition
+ * @author clabby <https://github.com/clabby>
+ * @notice This library contains helper functions for working with the `Position` type.
```

Comment:
How can i best verify the correctness of this? Is the spec for this documented someplace?

Reply:
No spec just yet, but can work on it. For the important pieces (attack / defend), here are some visuals:

![attack](https://github.com/ethereum-optimism/optimism/assets/8406232/28f45575-47b5-44d0-870a-064eb16dc860)
![defend](https://github.com/ethereum-optimism/optimism/assets/8406232/493e90a1-dd4c-41c0-8027-32bcdf2c0512)